### DISCUSSION TITLE
NullReferenceException: Object reference not set to an instance of an object

### DISCUSSION THREAD

**Discussion by zhoujie1998** [autocore-ai/MapToolbox#87] [General]
NullReferenceException: Object reference not set to an instance of an object
Packages.MapToolbox.IconManager.SetIconForObject (UnityEngine.GameObject obj, UnityEngine.Texture2D icon)

How can this be solved?

**Comment by Acoustic6** [comment]
I solved the problem by modifying the code of the SetIconForObject method in the IconManager class. What I did was replace the entire code of this method with EditorGUIUtility.SetIconForObject(obj, icon);. It appears that in some versions of EditorGUIUtility, SetIconForObject has become a public method and cannot be found using the specified flags (BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic).
